### PR TITLE
Reorganized instruction text on multiple challenges

### LINKED ISSUE #35908: Instruction Text mislabeled in challenges
I noticed that many of the challenges in the ["Introduction to the Coding Interview Data Structure Questions"](https://learn.freecodecamp.org/coding-interview-prep/data-structures) have the instruction text improperly placed in the challenge template. Instead of having them inside the `<section id= "instructions">`, they have been placed inside `<section id='description'>` and thus they appear differently than how instructions normally appear on the site (below the horizontal line). 

I have a pull request fix ready for this issue so I will submit it as soon as I get feedback regarding this.

### PR DESCRIPTION
**Pull request by AndrewZaw**
- [X ] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [X ] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X ] My pull request targets the `master` branch of freeCodeCamp.
- [X ] None of my changes are plagiarized from another source without proper attribution.
- [X ] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [ X] My changes do not use shortened URLs or affiliate links.

Closes #35908

### CODE REVIEW

**Review by RandellDawson — CHANGES_REQUESTED** [review]
@azaw502 Please remove the extra lines you have added below the headers and above `</section>` tags.  They are unnecessary.  Thank you.

**Comment by azaw502** [comment]
@RandellDawson spaces have been fixed

**Comment by RandellDawson** [comment]
@azaw502 I just looked and you still missed several extra lines which you added which still need to be removed.

**Comment by azaw502** [comment]
@RandellDawson I have looked them over and fixed them again.

File: curriculum/challenges/english/08-coding-interview-prep/data-structures/add-elements-at-a-specific-index-in-a-linked-list.english.md
Comment (RandellDawson) [inline curriculum/challenges/english/08-coding-interview-prep/data-structures/add-elements-at-a-specific-index-in-a-linked-list.english.md]:
```suggestion
<strong>Note:</strong> Remember to check if the given index is a negative or is longer than the length of the linked list.
```

File: curriculum/challenges/english/08-coding-interview-prep/data-structures/find-the-minimum-and-maximum-height-of-a-binary-search-tree.english.md
Comment (RandellDawson) [inline curriculum/challenges/english/08-coding-interview-prep/data-structures/find-the-minimum-and-maximum-height-of-a-binary-search-tree.english.md]:
```suggestion
<strong>Note:</strong> A similar property to height is depth, which refers to how far a given node is from the root node.
```

**Review by RandellDawson — CHANGES_REQUESTED** [review]
@azaw502 OK, this should be the last change requests from me.  Take care of these and then another collaborator will create a second review to make sure we have not missed anything.

**Review by RandellDawson — APPROVED** [review]
After all the changes requested changes have been committed by user, I think this PR is ready to be merged once an earlier PR (#35825) has been reviewed/merged first.

Keep in mind, once #35825 has been merged, there may be a few merge conflicts which will need to be resolved on this PR.

File: curriculum/challenges/english/08-coding-interview-prep/data-structures/add-elements-at-a-specific-index-in-a-linked-list.english.md
Comment (Manish-Giri) [inline curriculum/challenges/english/08-coding-interview-prep/data-structures/add-elements-at-a-specific-index-in-a-linked-list.english.md]:
```suggestion
Create an <code>addAt(index,element)</code> method that adds an element at a given index. Return false if an element could not be added.
```

File: curriculum/challenges/english/08-coding-interview-prep/data-structures/create-a-hash-table.english.md
Comment (Manish-Giri) [inline curriculum/challenges/english/08-coding-interview-prep/data-structures/create-a-hash-table.english.md]:
```js
var hash = string => {
  called++;
  var hash = 0;
```
Can we rename the local `hash` variable to something else? Like `hashed`? I don't see any reason to create a variable with the same name as the function. This might confuse beginners.

File: curriculum/challenges/english/08-coding-interview-prep/data-structures/create-a-queue-class.english.md
Comment (Manish-Giri) [inline curriculum/challenges/english/08-coding-interview-prep/data-structures/create-a-queue-class.english.md]:
```suggestion
Write an <code>enqueue</code> method that pushes an element to the tail of the queue, a <code>dequeue</code> method that removes and returns the front element, a <code>front</code> method that lets us see the front element, a <code>size</code> method that shows the length, and an <code>isEmpty</code> method to check if the queue is empty.
```

File: curriculum/challenges/english/08-coding-interview-prep/data-structures/create-a-hash-table.english.md
Comment (RandellDawson) [reply curriculum/challenges/english/08-coding-interview-prep/data-structures/create-a-hash-table.english.md]: It is a bit outside the scope of this PR, but I am fine with you committing such a change to this PR.

File: curriculum/challenges/english/08-coding-interview-prep/data-structures/create-a-trie-search-tree.english.md
Comment (Manish-Giri) [inline curriculum/challenges/english/08-coding-interview-prep/data-structures/create-a-trie-search-tree.english.md]: Use `<code>...</code>` tags for method names, like `add`, `isWord`, etc.

File: curriculum/challenges/english/08-coding-interview-prep/data-structures/create-a-trie-search-tree.english.md
Comment (Manish-Giri) [inline curriculum/challenges/english/08-coding-interview-prep/data-structures/create-a-trie-search-tree.english.md]: `<code>` tags for `Node`, `end`, etc.

File: curriculum/challenges/english/08-coding-interview-prep/data-structures/implement-heap-sort-with-a-min-heap.english.md
Comment (Manish-Giri) [inline curriculum/challenges/english/08-coding-interview-prep/data-structures/implement-heap-sort-with-a-min-heap.english.md]: `<code>` tags for `MinHeap` and method names.

**Review by Manish-Giri — DISMISSED** [review]
Please address the suggested changes.

**Comment by RandellDawson** [comment]
Hi @AndrewZaw ,

As there has been no response from your end for a while, we are closing this PR.

Feel free to request this PR to be reopened or create a new PR, if you still wish to make the changes you originally added plus any additional change requests asked of you.  Also, remember to update your fork, so you have the latest version of master in case it has changed significantly.

Happy Coding!

**Comment by moT01** [comment]
Maybe we should just give this a review and merge it as is or force push the changes we want @RandellDawson - after all, it does probably fix what is described in the issue. And even if it doesn't contain all the changes we want, it does probably improve the files as well.

**Comment by RandellDawson** [comment]
@Mo1914 I am fine with reopening it.  Will you make the changes?

**Review by RandellDawson — APPROVED** [review]
@Manish-Giri I have made all of your suggested changes (and added a few other code tags) on behalf of the user.  I think this LGTM.

**Review by moT01 — APPROVED** [review]
LGTM

**Review by moT01 — APPROVED** [review]
Thanks for contributing to freeCodeCamp @AndrewZaw! 🎉 

Thanks for making the requested changes @RandellDawson 🎉

**Review by Manish-Giri — APPROVED** [review]
LGTM 💯

**Pull request merged**
